python/tchannel/tornado/message_factory.py

```diff
@@ -315,10 +329,38 @@ def fragment(message):
                 payload_space = (common.MAX_PAYLOAD_SIZE -
                                  rw.length_no_args(message))
                 fragment_msg = message.fragment(payload_space)
+                self.generate_checksum(message, message_id)
                 yield message
         else:
             yield message
 
+    def generate_checksum(self, message, message_id):
+        if message.message_type not in CHECKSUM_MSG_TYPES:
+            return
+        generate_checksum(
+            message,
+            self.out_checksum.get(message_id, 0),
+        )
+
+        self.out_checksum[message_id] = message.checksum[1]
+        if message.flags == FlagsType.none:
+            self.out_checksum.pop(message_id)
+
+    def verify_message(self, message, message_id):
+        """Verify the checksum of the message."""
+        if verify_checksum(
+                message,
+                self.in_checksum.get(message_id, 0),
+        ):
+            self.in_checksum[message_id] = message.checksum[1]
+
+            if message.flags == FlagsType.none:
+                self.in_checksum.pop(message_id)
+        else:
+            if message_id in self.in_checksum:
+                self.in_checksum.pop(message_id)
+            raise InvalidChecksumException("Checksum does not match!")
```

Comment:
This is fine for now but a code-style nit for future changes:

Generally, instead of

    if foo:
        # do stuff
    else:
        raise Something

It's cleaner to do:

    if not foo:
        raise Something
    # do stuff